src/Redux/Features/Auth/LoginSlice.ts

```diff
@@ -0,0 +1,61 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { jwtDecode } from "jwt-decode";
+import { useEffect } from "react";
+
+interface AuthState {
+  authToken: string | null;
+  userRole: string | null; // Add userRole to AuthState
+  error: string | null;
+}
+
+const initialState: AuthState = {
+  authToken: null,
+  userRole: null,
+  error: null,
+};
+
+const authSlice = createSlice({
+  name: "auth",
+  initialState,
+  reducers: {
+    setAuthToken: (state, action: PayloadAction<string>) => {
+      state.authToken = action.payload;
+    },
+    setUserRole: (state, action: PayloadAction<string>) => {
+      state.userRole = action.payload;
+    },
+    setError: (state, action: PayloadAction<string>) => {
+      state.error = action.payload;
+    },
+    clearError: (state) => {
+      state.error = null;
+    },
+    logout: (state) => {
+      state.authToken = null;
+      state.userRole = null; // Clear userRole on logout
+    },
+  },
+});
+
+// Function to save user data including the user role
+export const saveUserData = () => {
+  const encodedToken: any = localStorage.getItem("authToken");
+  const decodedToken: any = jwtDecode(encodedToken);
+  console.log(decodedToken);
+  setUserRole(decodedToken);
+};
+
+export const { setAuthToken, setUserRole, setError, clearError, logout } =
+  authSlice.actions;
+
+export const requestHeaders = {
+  Authorization: `Bearer ${localStorage.getItem("authToken")}`,
```

Comment:
The token returns `Bearer token value`, so you need to remove `Bearer` here. When others use this headers,  it will be `Bearer Bearer tokenValue`.